src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerTopPanel.xaml

```diff
@@ -25,47 +27,136 @@
-        x:Name="_labelInstalled"
-        AutomationProperties.AutomationId="Tab_Installed"
-        Filter="Installed"
-        Margin="35,0,0,0"
-        Text="{x:Static nuget:Resources.Label_Installed}"
-        ControlSelected="FilterLabel_ControlSelected">
-      </nuget:FilterLabel>
-      <nuget:FilterLabel
-        Grid.Column="2"
-        x:Name="_labelUpgradeAvailable"
-        AutomationProperties.AutomationId="Tab_Updates"
-        Filter="UpdatesAvailable"
-        Margin="35,0,0,0"
-        Text="{x:Static nuget:Resources.Label_Updates}"
-        ControlSelected="FilterLabel_ControlSelected">
-      </nuget:FilterLabel>
-      <nuget:FilterLabel
-        Grid.Column="3"
-        x:Name="_labelConsolidate"
-        AutomationProperties.AutomationId="Tab_Consolidate"
-        Filter="Consolidate"
-        Margin="35,0,0,0"
-        Text="{x:Static nuget:Resources.Action_Consolidate}"
-        ControlSelected="FilterLabel_ControlSelected"
-        Visibility="Collapsed" />
+      <TabControl x:Name="tabsPackageManagement" IsSynchronizedWithCurrentItem="False" SelectionChanged="TabsPackageManagement_SelectionChanged">
+        <TabControl.Resources>
+          <Style TargetType="TabItem">
+            <Setter Property="Background" Value="{DynamicResource {x:Static nuget:Brushes.HeaderBackground}}" />
+            <Setter Property="FontSize" Value="{Binding FontSize, RelativeSource={RelativeSource AncestorType=UserControl},Converter={StaticResource Font122PercentSizeConverter}}"/>
+            <Setter Property="Foreground" Value="{DynamicResource {x:Static nuget:Brushes.UIText}}" />
+            <Setter Property="Padding" Value="20,0,20,0" />
+            <Setter Property="Template">
+              <Setter.Value>
+                <ControlTemplate TargetType="{x:Type TabItem}">
+                  <Grid x:Name="templateRoot" SnapsToDevicePixels="true">
+                    <Grid.RowDefinitions>
+                      <RowDefinition Height="auto" />
+                      <RowDefinition Height="auto" />
+                    </Grid.RowDefinitions>
+                    <Border x:Name="mainBorder" Background="{TemplateBinding Background}" BorderThickness="0" Margin="0">
+                    </Border>
+                    <ContentPresenter  Grid.Row="0" x:Name="contentPresenter" ContentSource="Header" Focusable="False" HorizontalAlignment="{Binding HorizontalContentAlignment, RelativeSource={RelativeSource AncestorType={x:Type ItemsControl}}}" Margin="{TemplateBinding Padding}" RecognizesAccessKey="True" SnapsToDevicePixels="{TemplateBinding SnapsToDevicePixels}" VerticalAlignment="{Binding VerticalContentAlignment, RelativeSource={RelativeSource AncestorType={x:Type ItemsControl}}}"/>
+                    <!-- the line under the text to indicate that this label is selected -->
+                    <Rectangle
+                      x:Name="_underline"
+                      Grid.Row="1"
+                      Margin="0"
```

Comment:
Margin default value is 0, so this is extraneous and (very) slightly wasteful.

Reply:
More perf wins!